### PR TITLE
(core) - Golf some bytes in typenames

### PR DESCRIPTION
**Pull request by JoviDeCroock**
This removes some redundant code from `core/utils/typenames.ts`

### CODE REVIEW

File: packages/core/src/utils/typenames.ts [hunk 1]
```diff
@@ -42,9 +41,7 @@ const hasTypenameField = (set: SelectionSetNode) => {
 };
 
 const formatNode = (node: FieldNode | InlineFragmentNode) => {
-  if (!node.selectionSet) {
-    return false;
```
Comment (kitten) [inline packages/core/src/utils/typenames.ts]: The return false here isn’t required I believe hit it does stop `visit` from traversing the rest of the AST. Not sure if that even makes a difference here though

File: packages/core/src/utils/typenames.ts [hunk 2]
```diff
@@ -20,11 +19,10 @@ const collectTypes = (obj: EntityLike | EntityLike[], types: string[] = []) => {
     });
   } else if (typeof obj === 'object' && obj !== null) {
     for (const key in obj) {
-      const val = obj[key];
-      if (key === '__typename' && typeof val === 'string') {
-        types.push(val);
+      if (key === '__typename' && typeof obj[key] === 'string') {
```
Comment (JoviDeCroock) [inline packages/core/src/utils/typenames.ts]: can typeof obj[key] even be no string when we are dealing with a __typename?

File: packages/core/src/utils/typenames.ts (hunk 1 above)
Comment (JoviDeCroock) [reply packages/core/src/utils/typenames.ts]: There isn't much to traverse further if it has no selectionSet so I think removing it is acceptable

File: packages/core/src/utils/typenames.ts (hunk 2 above)
Comment (kitten) [reply packages/core/src/utils/typenames.ts]: Technically not. We just haven’t made this assumption for core

Comment (JoviDeCroock) [reply packages/core/src/utils/typenames.ts]: let's jsut merge as is then :D

**Pull request merged**